### PR TITLE
(feat) add token metamask popup

### PR DESCRIPTION
**Pull request by karmabadger**
Made a new version of this

### CODE REVIEW

File: frontend/src/pages/Testing.tsx [hunk 1]
```diff
@@ -115,6 +116,24 @@ const Dashboard = () => {
...
         </div>
+
+        <div>
+          <button
+            onClick={async () => {
+              walletAddToken(
+                String(process.env.REACT_APP_PAYMENT_TOKEN_CONTRACT_ADDRESS),
+                String(process.env.REACT_APP_PAYMENT_TOKEN_NAME),
+                String(process.env.REACT_APP_PAYMENT_TOKEN_IMG_URL),
+                parseInt(String(process.env.REACT_APP_PAYMENT_TOKEN_DECIMALS))
```
Comment (deweller) [inline frontend/src/pages/Testing.tsx:127]: Please add all these to `.env.example`.

File: frontend/src/utils/metamask.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          image: tokenImgUrl, // A string url of the token logo
+        },
+      },
+    });
+
+    if (wasAdded) {
+      console.log('Added token', tokenSymbol, 'to wallet');
+    } else {
+      console.log('Cancelled adding', tokenSymbol, 'token to wallet');
+    }
```
Comment (deweller) [inline frontend/src/utils/metamask.ts:26]: Please remove these console logs.  We might use this function in the future in a production context.

**Review by deweller — CHANGES_REQUESTED** [review]

**Comment by karmabadger** [comment]
I have made the requested changes

File: frontend/src/utils/metamask.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      params: {
+        type: 'ERC20', // Initially only supports ERC20, but eventually more!
+        options: {
+          address: tokenAddress, // The address that the token is at.
+          symbol: tokenSymbol, // A ticker symbol or shorthand, up to 5 chars.
+          decimals: tokenDecimals, // The number of decimals in the token
+          image: tokenImgUrl, // A string url of the token logo
+        },
+      },
+    });
```
Comment (deweller) [inline frontend/src/utils/metamask.ts:20]: Please remove wasAdded.

You can find these types of errors with `yarn run lint`
```
src/utils/metamask.ts
--
10:46:26.446 | Line 9:11:  'wasAdded' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

**Review by deweller — CHANGES_REQUESTED** [review]

File: frontend/src/utils/metamask.ts (hunk 3 above)
Comment (karmabadger) [reply frontend/src/utils/metamask.ts:20]: hmmm this variable can be useful in an if else statement if needed to check if the token was correctly added tho.

Comment (karmabadger) [reply frontend/src/utils/metamask.ts:20]: but I'll still remove it and leave a comment there.

Comment (karmabadger) [reply frontend/src/utils/metamask.ts:20]: fixed

**Review by deweller — DISMISSED** [review]

**Review by deweller — APPROVED** [review]

**Pull request merged**